server/src/main/java/org/opensearch/node/remotestore/RemoteStoreNodeAttribute.java

```diff
@@ -145,12 +141,36 @@ private RepositoriesMetadata buildRepositoriesMetadata(DiscoveryNode node) {
         return new RepositoriesMetadata(repositoryMetadataList);
     }
 
+    private Set<String> getValidatedRepositoryNames(DiscoveryNode node) {
+        Set<String> repositoryNames = new HashSet<>();
+        if (node.getAttributes().containsKey(REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY)
+            || node.getAttributes().containsKey(REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY)) {
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_STORE_SEGMENT_REPOSITORY_NAME_ATTRIBUTE_KEY));
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_STORE_TRANSLOG_REPOSITORY_NAME_ATTRIBUTE_KEY));
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY));
+        } else if (node.getAttributes().containsKey(REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY)) {
+            repositoryNames.add(validateAttributeNonNull(node, REMOTE_STORE_CLUSTER_STATE_REPOSITORY_NAME_ATTRIBUTE_KEY));
+        }
+        return repositoryNames;
+    }
+
     public static boolean isRemoteStoreAttributePresent(Settings settings) {
```

Comment:
Do we still need this method? Caller can explicitly call `isRemoteDataAttributePresent` or `isRemoteClusterStateAttributePresent`.

Reply:
This method is used being here. 
https://github.com/opensearch-project/OpenSearch/blob/4a0feeef41fd0efee342d31e5e935c31af332cf9/server/src/main/java/org/opensearch/node/Node.java#L1896-L1898

This method simply check if any remote store attribute is present.